pkg/apis/v1alpha1/vpcnetworkconfiguration_types.go

```diff
@@ -0,0 +1,120 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// SubnetType describes the subnet type.
+type SubnetType string
+
+const (
+	SubnetTypePublic  SubnetType = "public"
+	SubnetTypePrivate SubnetType = "private"
+)
+
+// VPCNetworkConfigurationSpec defines the desired state of VPCNetworkConfiguration
+// There is a default VPCNetworkConfiguration that applies to Namespaces
+// do not have a VPCNetworkConfiguration assigned.
+// When a field is not set in a Namespace's VPCNetworkConfiguration,
+// the Namespace will use the value in the default VPCNetworkConfiguration.
+type VPCNetworkConfigurationSpec struct {
+	// Namespaces to be applied.
+	// If not set, the default VPCNetworkConfiguration applying to all Namespaces.
+	AppliedToNamespaces []string `json:"appliedToNamespaces,omitempty"`
+	// Tier0 gateway or Tier0 VRF path.
+	Tier0Gateway string `json:"tier0Gateway,omitempty"`
+	// Edge cluster path on which the networking elements will be created.
+	EdgeCluster string `json:"edgeCluster,omitempty"`
+	// NSX-T Project the Namespace associated with.
+	NSXTProject string `json:"nsxtProject,omitempty"`
+	// NSX-T IP Block paths used to allocate public Subnets.
+	// +kubebuilder:validation:MinItems=0
+	// +kubebuilder:validation:MaxItems=5
+	PublicIPV4Block []string `json:"publicIPV4Block,omitempty"`
```

Comment:
Should it be "IPV4" or "IPv4"? I think should be "IPv4".

Reply:
changed to IPv4. Thanks.